tests/sha3.cc

```diff
@@ -52,30 +52,80 @@ class Sha3KAT : public ::testing::TestWithParam<TestCase>
 
 TEST_P(Sha3KAT, TryKAT)
 {
-  const TestCase& test_case(GetParam());
-
-  // Stupid const
-  uint8_t* msg = const_cast<uint8_t*>(test_case.msg.data());
-
-  bytes digest(test_case.md.size(), 0);
-  if (test_case.md.size() == 32) {
-    Hacl_SHA3_sha3_256(test_case.msg.size(), msg, digest.data());
-  } else if (test_case.md.size() == 48) {
-    Hacl_SHA3_sha3_384(test_case.msg.size(), msg, digest.data());
-  } else if (test_case.md.size() == 64) {
-    Hacl_SHA3_sha3_512(test_case.msg.size(), msg, digest.data());
+  auto test_case = GetParam();
+
+  {
+    bytes digest(test_case.md.size(), 0);
+    if (test_case.md.size() == 224 / 8) {
+      Hacl_SHA3_sha3_224(
+        test_case.msg.size(), test_case.msg.data(), digest.data());
+    } else if (test_case.md.size() == 256 / 8) {
+      Hacl_SHA3_sha3_256(
+        test_case.msg.size(), test_case.msg.data(), digest.data());
+    } else if (test_case.md.size() == 384 / 8) {
+      Hacl_SHA3_sha3_384(
+        test_case.msg.size(), test_case.msg.data(), digest.data());
+    } else if (test_case.md.size() == 512 / 8) {
+      Hacl_SHA3_sha3_512(
+        test_case.msg.size(), test_case.msg.data(), digest.data());
+    }
+
+    EXPECT_EQ(test_case.md, digest) << bytes_to_hex(test_case.md) << std::endl
+                                    << bytes_to_hex(digest) << std::endl;
   }
 
-  EXPECT_EQ(test_case.md, digest) << bytes_to_hex(test_case.md) << std::endl
-                                  << bytes_to_hex(digest) << std::endl;
+  // TODO: Evercrypt
```

Comment:
Is SHA3 in evercrypt? In any case, either add an issue number or drop if it's not needed.

Reply:
Should it be in EverCrypt?

Reply:
:+1: